test/unit/serialization/compact/CompactTest.js

```diff
@@ -244,4 +246,18 @@ describe('CompactTest', function () {
         const object = await serializationService.toObject(data);
         object.should.be.deep.equal(employer);
     });
+
+    it('should not serialize without nested fields serializer', async function() {
```

Comment:
```suggestion
    it('should throw proper error when nested field serializer is missing', async function() {
```

Reply:
The main problem was throwing TypeError in an internal code of the Node.js client. Now we handle it and throw a proper messaged error. Hence the name suggestion.

Reply:
Fixed at https://github.com/hazelcast/hazelcast-nodejs-client/pull/1365/commits/3345e5697bb3247845774bd13ae536ebe3124fc2.